libr/debug/p/native/linux/linux_debug.c

```diff
@@ -491,15 +504,35 @@ static bool linux_kill_thread(int tid, int signo) {
 static bool linux_stop_thread(int tid) {
 	int status;
 	int ret = -1;
+
 	if (linux_kill_thread (tid, SIGSTOP)) {
-		if ((ret = waitpid (tid, &status, __WALL)) == -1) {
+		if ((ret = waitpid (tid, &status, WUNTRACED)) == -1) {
 			perror("waitpid");
 		}
 	}
 
 	return ret == tid;
 }
 
+bool linux_stop_threads(RDebug *dbg, int except) {
+	// Stop all currently debugged threads
+	RList *list = dbg->threads;
+	bool ret = true;
+	if (list) {
+		RDebugPid *th;
+		RListIter *it;
+		r_list_foreach (list, it, th) {
+			if (th->pid && th->pid != except) {
+				if (!linux_stop_thread (th->pid)) {
```

Comment:
extra `if`, can be combined

Reply:
i think it's fine like this. it's more readable

Reply:
It would print an unnecessary error if th->pid == except or if th->pid was undefined.